cobbler/items/abstract/inheritable_item.py

```diff
@@ -0,0 +1,339 @@
+"""
+"InheritableItem" the entry point for items that have logical parents and children.
+
+Changelog:
+    * V3.4.0 (unreleased):
+        * Initial creation of the class
+"""
+
+# SPDX-License-Identifier: GPL-2.0-or-later
+# SPDX-FileCopyrightText: Enno Gotthold <enno.gotthold@suse.com>
+
+from abc import ABC
+from typing import TYPE_CHECKING, Any, Dict, List, Optional, Tuple, Union
+
+from cobbler.cexceptions import CX
+from cobbler.decorator import LazyProperty
+from cobbler.items.abstract.base_item import BaseItem
+
+if TYPE_CHECKING:
+    from cobbler.api import CobblerAPI
+    from cobbler.cobbler_collections.collection import ITEM, ITEM_UNION
+    from cobbler.items.distro import Distro
+    from cobbler.items.menu import Menu
+    from cobbler.items.profile import Profile
+    from cobbler.items.system import System
+    from cobbler.settings import Settings
+
+
+class InheritableItem(BaseItem, ABC):
+    """
+    Abstract class that acts as a starting point in the inheritance for items that have parents and children.
+    """
+
+    # Constants
+    TYPE_NAME = "inheritable_abstract"
+    COLLECTION_TYPE = "inheritable_abstract"
+
+    # Item types dependencies.
+    # Used to determine descendants and cache invalidation.
+    # Format: {"Item Type": [("Dependent Item Type", "Dependent Type attribute"), ..], [..]}
+    TYPE_DEPENDENCIES: Dict[str, List[Tuple[str, str]]] = {
```

Comment:
Could we take the opportunity to refactor these two data structures (`TYPE_DEPENDENCIES` and `LOGICAL_INHERITANCE`) to make them a bit more readable?

My problem is that the inner tuples, e.g. `("profile", "repos")`, rely on readers to remember what the first and second member means (or, to refer to the clarifying comment above, and/or refer to the tests)
 
We could fix that by using data classes, though they were added in Python 3.7 IIRC. 

For a solution compatible with Python 3.6, we can use a `namedtuple`, e.g.:

```
# Define Named Tuple
HierarchyItem = namedtuple("HierarchyItem", "child_class parent_attribute parent_class")

# Python 3.6 compatible defaults setting, 
# so that we can reuse one namedtuple for both structures
HierarchyItem.__new__.__defaults__ = (None, None, None)
```

This tuple could then be reused in both of these structures, e.g.:

```
    TYPE_DEPENDENCIES: Dict[str, List[Tuple[str, str]]] = {
        "repo": [
            HierarchyItem(child_class="profile", parent_attribute="repos"),
        ],
...
```
or

```
    LOGICAL_INHERITANCE: Dict[
        str, Tuple[List[Tuple[str, str]], List[Tuple[str, str]]]
    ] = {
        "distro": (
            [],
            [
                HierarchyItem(child_class="profile", parent_class="distro"),
            ],
        ),
...
```
Then, we can access the values by name, e.g. `item.child_class` instead of `item[0]`, which to me makes these more understandable at a glance.

The downside is that each tuple definition is now longer of course, but the added verbosity makes it more readable. 

Note that since in the example, we're re-using one namedtuple class, we cannot rely on positional arguments, e.g. `HierarchyItem("..", "..")` - this would work for `TYPE_DEPENDENCIES`, but would cause `LOGICAL_INHERITANCE` to use `HierarchyItem("..", None, "..")` which is a bit ugly.

Reply:
I will force push in a few minutes where this then will be fixed.